### PR TITLE
Alfred cheat commit6 — updates options.py

### PR DESCRIPTION
**Pull request by odapg**
Updates options.py to show the cheatsheet's name between brackets for each entry when doing a global term search (maybe this should be optional)

### CODE REVIEW

**Comment by wx-Yao** [comment]
Let's use this PR for the review. As long as you confirm it has all the features you wanna add, other PRs can be ignored.

File: lib/options.py [hunk 1]
```diff
@@ -12,49 +13,99 @@ def __init__(self, parser, workflow):
...
-                Options.warning("Cheat sheet not found.","", self._workflow)
+                Options.warning("Cheatsheet not found.","", self._workflow)
                 return None
             ret=self._parser.searchInSheet(keyword, sheetName, self._workflow)
+            html_file = html_gen.create_html_global_search(ret,keyword,sheetName)
+            tmp_filename = self._workflow.cachedir + "/" + sheetName +'.html'
         if ret==[]:
             Options.warning("Not found", "No match found for search {}".format(keyword), self._workflow)
             return None
+# Create HTML file
```
[the remark this answers is not in the document]
Comment (odapg) [reply lib/options.py:28]: OK!

File: my_style.css [hunk 2]
```diff
@@ -0,0 +1,61 @@
+html { font-size: 100%; overflow-y: scroll; }
```
[the remark this answers is not in the document]
Comment (odapg) [reply my_style.css:1]: Sure, if you have ideas, that would be cool :-)

**Comment by wx-Yao** [comment]
@odapg hey man, since the thread has been silent for a while, and I do like some of your new features, I'll merge the PR and make some code change to either fix something or change how some of your feature present (e.g., the `--search` is a bit of weird. Why don't we just type to search :) ). And then commit again. The result might not be exactly what you expected but very close. Is that ok for you?

**Comment by odapg** [comment]
Hey Wayne,
Sorry for my late answers, I have been away for some time.Yes, what you propose is perfectly OK to me!


    Le samedi 11 juillet 2020 à 06:14:20 UTC+2, Wayne Yao <notifications@github.com> a écrit :  
 
 


@odapg hey man, since the thread has been silent for a while, and I do like some of your new features, I'll merge the PR and make some code change to either fix something or change how some of your feature present (e.g., the --search is a bit of weird. Why don't we just type to search :) ). And then commit again. The result might not be exactly what you expected but very close. Is that ok for you?

**Comment by wx-Yao** [comment]
@odapg Hi.

I go through all your implementation today and have a couple of comments.

1. Your implementation has a bug that global search doesn't work. 
image

2. In your forked repo, I notice this:

> New!
> - You can add hidden blocks in your file that will not appear in Alfred nor in the Quicklook view: simply begin the block by /*(no need to close the block)
> - You can add sections that will not appear in Alfred but that will appear in the Quicklook view. Begin the block by //.

Though I understand you wanna make it feature-rich and flexible in terms of writing cheat sheet, I think this pattern complicates the original design. We have 3 types of comments now, the "#", the "/**" and the "//", I had to remember what type of comment means what when writing some simple notes. That experience is not optimal.

This pattern also explains the complexity of your implementation. The code becomes a bit of hard to read and maintain.

That said, I still appreciate your ideas and the new proposed features are good. How about this:

1. I'm afraid I couldn't merge your PR as I had some problem understanding it (hence improve it). But

2. Your ideas are precious. Let me implement them. You'll be listed as a contributor in the readme.

Features being implemented:

1. Large text preview.

2. multiple lines commands and comments.

3. Open with editor

4. The preview in HTML. But when writing sheet, "#" means normal comments, "##" means comments not visible in Alfred but visible in preview HTML. That makes it simple.

How does it sound to you?

Thanks

**Pull request closed without merging**